pkg/container-collection/tracerconnect/connect.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracerconnect
+
+import (
+	"fmt"
+	"os"
+
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	containerutils "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type NetworkingTracer[Event any] interface {
+	Attach(key string, pid uint32, eventCallback func(Event)) error
+	Detach(key string) error
+}
+
+type ConnectNetworkingTracerToContainerCollectionArgs[Event any] struct {
+	Key           string
+	Tracer        NetworkingTracer[Event]
+	Resolver      containercollection.ContainerResolver
+	Selector      containercollection.ContainerSelector
+	EventCallback func(*containercollection.Container, Event)
+	Base          func(eventtypes.Event) Event
+}
+
+// ConnectNetworkingTracerToContainerCollection connects a networking
+// tracer to the container collection package. It creates the needed
+// logic to call the Attach() function on the tracer each time a
+// container is created and to call Detach() each time the container is
+// removed. Callers must call args.Resolver.Unsubscribe(key) to remove
+// the connection.
+func ConnectNetworkingTracerToContainerCollection[Event any](
+	args *ConnectNetworkingTracerToContainerCollectionArgs[Event],
+) error {
+	return connectNetworkingTracerToContainerCollectionInternal(
+		args.Key, args.Tracer, args.Resolver, args.Selector, args.EventCallback,
+		args.Base,
+	)
+}
+
+func connectNetworkingTracerToContainerCollectionInternal[Event any](
+	key string,
+	tracer NetworkingTracer[Event],
+	resolver containercollection.ContainerResolver,
+	selector containercollection.ContainerSelector,
+	eventCallback func(*containercollection.Container, Event),
+	base func(eventtypes.Event) Event,
+) error {
+	netnsHost, err := containerutils.GetNetNs(os.Getpid())
+	if err != nil {
+		return fmt.Errorf("getting network namespace: %w", err)
+	}
+	genKey := func(container *containercollection.Container) string {
+		if container.Netns == netnsHost {
+			return "host"
+		}
+
+		key := container.Namespace + container.Podname
+		// for local-gadget as container.Namespace and
+		// container.Podname can be empty
+		if key == "" {
+			key = container.ID
+		}
+
+		return key
```

Comment:
Variable shadowing: could the variable be named `attachKey` so it is not confused with the `subscribeKey` variable from line 56?

The function could be named `genAttachKey()`.